Gems/PhysX/Code/Source/Scene/PhysXScene.h

```diff
@@ -79,7 +79,27 @@ namespace PhysX
 
         physx::PxControllerManager* GetOrCreateControllerManager();
 
+        //! Apply batched transform sync events for the current simulation pass. 
+        //! This will clear the batched data for the next simulation pass.
+        void FlushTransformSync();
+        
     private:
+
+        //! Data structure for efficient unique vector functionality.
+        //! Body indices are inserted avoiding duplicated data and stored in a vector for efficient iteration.
+        class QueuedActiveBodyIndices
+        {
+        public:
+            void Insert(AzPhysics::SimulatedBodyIndex bodyIndex);
+            void Reserve(size_t reserveSize);
+            void Clear();
+            void Apply(const AZStd::function<void(AzPhysics::SimulatedBodyIndex)>& applyFunction);
+
+        private:
+            AZStd::unordered_set<AzPhysics::SimulatedBodyIndex> m_set;
+            AZStd::vector<AzPhysics::SimulatedBodyIndex> m_array;
```

Comment:
suggestion: It might be nicer to name these members after what they are and not the types, so this could be `m_uniqueIndices` for the set and `m_packedIndices` for the vector? Not a huge deal but generally tends to make the code more resilient if the types change in future.

Reply:
Nice names, I'll use them, thanks!

Reply:
Awesome thank you!